server/src/test/java/org/elasticsearch/env/NodeEnvironmentTests.java

```diff
@@ -470,6 +471,72 @@ public void testExistingTempFiles() throws IOException {
         }
     }
 
+    public void testFailOnDataForNonDataNode() throws IOException {
+        Settings settings = buildEnvSettings(Settings.EMPTY);
+        final NodeEnvironment env = newNodeEnvironment(settings);
```

Comment:
`buildEnvSettings` is called as part of `newNodeEnvironment`. Just calling `newNodeEnvironment()` here is sufficient.

Reply:
The reason I explicitly build settings here is to be able to reuse the path.node setting for the next newNodeEnvironment call. Is there a better way?